### ISSUE TITLE
Interrupt speech when a new window or dialog box is opened automatically or a focused content within a dialog box changes

### ISSUE THREAD

**Issue by winman3000** [nvaccess/nvda#14010]
### Is your feature request related to a problem? Please describe.

It is not really a problem. In many dialog boxes it is the case that the speech is not interrupted when information changes, such as progress information or a list that is constantly updated. Sometimes it also happens that applications automatically open a window. NVDA only reads this information when the older information has finished speaking. In this case it can take longer until the users get the newest information.

### Describe the solution you'd like

It would be very nice if speech is interrupted as soon as:
* A new dialog box opens
* A new window opens
* Contents in a dialog box are changed or updated

It would also be conceivable to make this option optional. Users who prefer the old behavior can turn this feature off.

### Describe alternatives you've considered

Basically, there are no alternatives except to constantly look at the Braille display or press the CTRL key to manually check if the information has changed. You could of course wait until NVDA has spoken everything, but this takes too much time for me personally.

**Comment by cary-rowen** [comment]
Does the new popup dialog have the highest priority? If I'm concentrating on reading, a pop-up dialog box will interrupt my train of thought.

**Comment by winman3000** [comment]
Since a popup or similar would disturb my attention anyway, even if it is delayed, I wouldn't care. However, since there are different opinions about this, this feature should be optionally switchable on or off, as I said.

**Comment by lukaszgo1** [comment]
@winman3000 Please give specific examples of situations in which NVDA does not interrupt speech as expected - as it stands someone wanting to work on the issue has no clear STR to follow.

**Comment by winman3000** [comment]
I can't think of any examples offhand. Such situations can occur in virtually any application where dialog boxes are updated more frequently or new windows are opened. As soon as I have a concrete example, I will of course describe it here.

**Comment by Brian1Gaff** [comment]
I have to say, that I do prefer the current behaviour, as there are so many 
misdesigned websites that say, now showing slide 2 of 5 or whatever every so 
often. These are of course pictures that cannot be heard, so stopping every 
time is a pain.
 It does seem that web designers do not know how to fix the problem with 
landmarks etc in their pages, as I think Amazon do.

 Brian

-- 
***@***.***
Sent via blueyonder.(Virgin media)
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
----- Original Message ----- 
From: "winman3000" ***@***.***>
To: "nvaccess/nvda" ***@***.***>
Cc: "Subscribed" ***@***.***>
Sent: Saturday, August 13, 2022 10:54 AM
Subject: [nvaccess/nvda] Interrupt speech when a new window or dialog box is 
opened automatically or a focused content within a dialog box changes (Issue 
#14010)

**Comment by Adriani90** [comment]
@winman3000 can you provide a certain use case to test on? Actually in the advanced settings of NVDA you can enable cancelable speech when the focus event expired, so this setting should already actually interupt old speech when the focus changes to a new element. Is this not the case for you?

**Comment by Adriani90** [comment]
We didn't get any updates from the initial author since more than one year, i am closing as abandoned. if you are still having this issue, please provide the requested info and we can reopen.
The only situation where I saw something like this was on a portable version of NVDA but this is covered in #13343.
